theano/sandbox/gpuarray/tests/test_basic_ops.py

```diff
@@ -340,30 +341,29 @@ def test_gpu_contiguous():
     assert f(a_val, 2).flags.c_contiguous
 
 
-class G_reshape(T_reshape):
+class G_reshape(test_basic.T_reshape):
     def shortDescription(self):
         return None
 
     def __init__(self, name):
-        T_reshape.__init__(self, name,
-                           shared=gpuarray_shared_constructor,
-                           op=GpuReshape,
-                           mode=mode_with_gpu,
-                           # avoid errors with limited devices
-#                             dtype='float32',
-                           ignore_topo=(HostFromGpu, GpuFromHost,
-                                        theano.compile.DeepCopyOp,
-                                        theano.sandbox.gpuarray.elemwise.GpuElemwise,
-                                        theano.tensor.opt.Shape_i,
-                                        theano.tensor.opt.MakeVector))
+        test_basic.T_reshape.__init__(
+            self, name,
+            shared=gpuarray_shared_constructor,
+            op=GpuReshape,
+            mode=mode_with_gpu,
+            ignore_topo=(HostFromGpu, GpuFromHost,
+                         theano.compile.DeepCopyOp,
+                         theano.sandbox.gpuarray.elemwise.GpuElemwise,
+                         theano.tensor.opt.Shape_i,
+                         theano.tensor.opt.MakeVector))
         assert self.op == GpuReshape
 
 
-class G_Join_and_Split(T_Join_and_Split):
+class G_Join_and_Split(test_basic.T_Join_and_Split):
     def setUp(self):
         super(G_Join_and_Split, self).setUp()
         self.mode = mode_with_gpu.excluding('constant_folding')
-        self.join_op = GpuJoin
+        self.join_op = GpuJoin()
         self.split_op = GpuSplit
         # Use join instead of MakeVector since there is no MakeVector on GPU
         self.make_vector_op = GpuJoin
```

Comment:
It is kind of confusing to have some Op classes and some Op instances, all named `..._op`.
I understand that we cannot pre-instantiate `GpuSplit` because the number of splits is a constructor argument, but we should name the members differently then.